### DISCUSSION TITLE
Need help using the indent command

### DISCUSSION THREAD

**Discussion by LorNtz** [prettier/prettier#14854] [Q&A] [answered]
I'm creating a plugin to format some javascript code with custom indentation and I'm getting started by making a demo. For example I have a snippet like this: 
```javascript
function     func (    ) {
              return null
}
func()
```
(looks ugly because I want to make sure that I'm leveraging the default printer)
I want the function call statement `func()` to indent by an extra level. After formatting I should get the following output:
```javascript
function func () {
  return null
}
  func()
```
So I came up with some code like this:
```javascript
const babelParsers = require("prettier/parser-babel").parsers
const {
  doc: {
    builders: {
      indent,
    },
  },
} = require('prettier')

let defaultPrint = null

function myPrint (path, options, print) {
  if (!defaultPrint) {
    const jsPlugin = options.plugins.find(
      (plugin) => plugin.printers && plugin.printers.estree
    )
    defaultPrint = jsPlugin.printers.estree.print
  }
  const node = path.getValue()
  let doc = defaultPrint(path, options, print)
  if (node && node.type === 'CallExpression') {
    console.log('calling indent command')
    doc = indent(doc)
  }
  return doc
}

module.exports = {
  languages: [
    {
      name: 'javascript',
      parsers: ['babel'],
      extensions: ['.js']
    }
  ],
  parsers: {
    babel: {
      ...babelParsers.babel,
      astFormat: 'babel-ast'
    }
  },
  printers: {
    'babel-ast': {
      print: myPrint,
    }
  }
}
```
But I'm not getting the `func()` indented as I expected. What I got is:
```javascript
function func () {
  return null
}
func()
```
Formatted, but no extra indentation for `func()`. I'm pretty sure that the `indent()` command was called, since I got 'calling indent command' printed out in my console. I have no idea whether I'm doing wrong with `indent()`. How could I fix the problem?

**Comment by fisker** [comment, marked as answer]
```diff
- doc = indent(doc)
+ doc = indent([line, doc])
```

**Reply by LorNtz** [reply]
Thanks for the help! Btw, I'm curious where can I get this info or guidelines from? The doc entry in `commands.md` only tells me that `indent()` receives a doc and returns an indented doc, and I don't even know there has to be a leading `line` command.
